janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -7409,4 +7409,20 @@ public void testVerticesDropAfterWhereWithBatchQueryEnabled() {
         assertFalse(tx.traversal().V().has("a", timestamp).has("b", true).has("c", true).bothE().hasNext());
     }
 
+    @Test
+    public void testPropertiesAfterProjectWithByThenChoose() {
```

Comment:
Nitpick: Can you add `withBatchQueryEnabled` or something similar to the test name, so that people know this test aims to test batch functionality?